pkg/kubelet/prober/prober.go

```diff
@@ -236,6 +238,11 @@ func (eic *execInContainer) Start() error {
 	if eic.writer != nil {
 		// only record the write error, do not cover the command run error
 		if p, err := eic.writer.Write(data); err != nil {
+			if errors.Is(err, io.ErrShortWrite) {
+				if lw, ok := eic.writer.(*ioutils.LimitedWriter); ok && int64(len(data)) > lw.N {
+					err = fmt.Errorf("%w: truncated output, please shorten the probe output", err)
+				}
+			}
```

Comment:
Would it make sense to return more descriptive error from the `LimitedWriter.Write` instead of doing error type checking and writer casting here?

Reply:
I modified it to truncate the output before writing and output a more specific error message

Reply:
Truncation is already done by `LimitedWriter.Write`. You may want to return a proper error from it as I've suggested above. This way you'd not need to cast writer here.

Reply:
Yes, Do you have any suggestions on how to do that?

Reply:
Sure. Return meaningful error instead of `ErrShortWrite` from `LimitedWriter.Write`.

BTW, `ErrShortWrite` description says that this error is returned when writer `failed to return an explicit error.`, which is another argument in favour of returning an explicit error.